Sources/Toolbox/Utilities/FailableCodable.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  FailableCodable.swift
+//  
+//
+//  Created by Oliver Krakora on 06.11.20.
+//
+
+import Foundation
+
+/// A helper that allows you to wrap potential faulty `Codable` data structures
+///
+/// **Usage**
+///
+///     struct Person: Codable {
+///         let name: String
+///         let websiteURL: FailableCodable<URL>
+///     }
+///
+/// The `websiteURL` is defined by a user so there is no guarantee that it is a valid `URL`.
+///
+///
+/// Without `FailableCodable` the decoding would fail for the entire `Person` and you would have to implement the decoding intializer yourself.
+///
+enum FailableCodable<T: Codable> {
+    /// Indicates that decoding of the underlying data was successfull
+    case success(T)
+    /// Indicates that the underlying data failed to decode
+    case error(Error)
+    
+    /// If the decoding was successfull a non nil value of type `T`
+    var value: T? {
+        switch self {
+        case .success(let value):
+            return value
+        case .error:
+            return nil
```

Comment:
switch can be replaces with:
```
guard case .error(let error) = self else { return nil }
```

Reply:
finde den switch hier aber besser (lesbarer)